.github/workflows/capgen_unit_tests.yaml

```diff
@@ -0,0 +1,20 @@
+name: Capgen Unit Tests
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+  push:
+    branches:
+      #Trigger workflow on push to any branch or branch heirarchy:
+      - '**'
+
+jobs:
+  unit_tests:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: build Docker image
+      run: docker build -f test/Dockerfile -t ccpp .
+    - name: run tests in container
+      run: docker run --name test-container -t ccpp bash -c 'source ccpp-env/bin/activate && cd ccpp/test && ./run_tests.sh'
```

Comment:
The more I think about NOAA's lukewarm response to containers for the framework, the more I wonder if it would make sense to just directly use the Github provided `ubuntu-latest` virtual environment, and use `sudo apt-get` to install anything needed that we don't have already in that environment.  That way we might have an easier time when we get to the point of unifying the NOAA and NCAR versions of the framework (which will also mean unifying our testing workflows), and we also might get less pushback when trying to get the capgen branch PR accepted and merged.

Reply:
Understood, updated to move everything into the `ubuntu-latest` container.  If it means better integration with everyone, I'm all on board with having one way to build/run instead of bifurcating environments.